### PR TITLE
Refactor

### PR DESCRIPTION
**Pull request by stevenewald**

### CODE REVIEW

File: exchange/src/client_manager/client_manager.cpp [hunk 1]
```diff
@@ -3,58 +3,67 @@
 namespace nutc {
 namespace manager {
 
-inline bool
-ClientManager::user_exists(const std::string& uid) const
+bool
+ClientManager::user_exists(const std::string& id) const
 {
-    return clients.find(uid) != clients.end();
+    return clients.find(id) != clients.end();
```
Comment (egelja) [inline exchange/src/client_manager/client_manager.cpp:9]:
```suggestion
    return clients.contains(id);
```
https://en.cppreference.com/w/cpp/container/unordered_map/contains
this should probably go into the header too

File: exchange/src/client_manager/client_manager.cpp [hunk 2]
```diff
@@ -3,58 +3,67 @@
...
 }
 
-inline bool
-ClientManager::user_holds_stock(const std::string& uid, const std::string& ticker) const
+void
+ClientManager::set_client_pid(const std::string& id, pid_t pid)
```
Comment (egelja) [inline exchange/src/client_manager/client_manager.cpp:13]: also should probably go into the header

File: exchange/src/client_manager/client_manager.cpp [hunk 3]
```diff
@@ -3,58 +3,67 @@
...
+
+bool
+ClientManager::user_holds_stock(const std::string& id, const std::string& ticker) const
 {
-    if (!user_exists(uid)) [[unlikely]]
+    if (!user_exists(id)) [[unlikely]]
         return false;
 
-    return clients.at(uid).holdings.find(ticker) != clients.at(uid).holdings.end();
+    return clients.at(id).holdings.find(ticker) != clients.at(id).holdings.end();
```
Comment (egelja) [inline exchange/src/client_manager/client_manager.cpp:27]:
```suggestion
    return clients[id].holdings.contains(ticker);
```

also should go into the header

File: exchange/src/client_manager/client_manager.cpp [hunk 4]
```diff
@@ -3,58 +3,67 @@
...
 float
-ClientManager::get_holdings(const std::string& uid, const std::string& ticker) const
+ClientManager::get_holdings(const std::string& id, const std::string& ticker) const
 {
-    if (!user_holds_stock(uid, ticker))
+    if (!user_holds_stock(id, ticker))
         return 0.0f;
 
-    return clients.at(uid).holdings.at(ticker);
+    return clients.at(id).holdings.at(ticker);
```
Comment (egelja) [inline exchange/src/client_manager/client_manager.cpp:36]:
```suggestion
    return clients[id].holdings[ticker];
```

also should go into the header

File: exchange/src/client_manager/client_manager.cpp [hunk 5]
```diff
@@ -3,58 +3,67 @@
...
-    clients[uid].holdings[ticker] += change_in_holdings;
+    clients[id].holdings[ticker] += change_in_holdings;
 }
 
 std::optional<messages::SIDE>
 ClientManager::validate_match(const messages::Match& match) const
 {
     float trade_value = match.price * match.quantity;
-    bool insufficient_capital = get_capital(match.buyer_uid) - trade_value < 0;
+    bool insufficient_capital = get_capital(match.buyer_id) - trade_value < 0;
```
Comment (egelja) [inline exchange/src/client_manager/client_manager.cpp:59]:
```suggestion
    float remaining_capital = get_capital(match.buyer_id) - trade_value;
```

File: exchange/src/client_manager/client_manager.cpp [hunk 6]
```diff
@@ -3,58 +3,67 @@
...
     bool insufficient_holdings =
```
Comment (egelja) [inline exchange/src/client_manager/client_manager.cpp:60]:
```suggestion
    bool remaining_holdings =
```

File: exchange/src/client_manager/client_manager.cpp [hunk 7]
```diff
@@ -3,58 +3,67 @@
...
-        get_holdings(match.seller_uid, match.ticker) - match.quantity < 0;
+        get_holdings(match.seller_id, match.ticker) - match.quantity < 0;
```
Comment (egelja) [inline exchange/src/client_manager/client_manager.cpp:61]:
```suggestion
        get_holdings(match.seller_id, match.ticker) - match.quantity;
```

File: exchange/src/client_manager/client_manager.cpp [hunk 8]
```diff
@@ -3,58 +3,67 @@
...
 
     if (insufficient_capital) [[unlikely]]
```
Comment (egelja) [inline exchange/src/client_manager/client_manager.cpp:63]:
```suggestion
    if (remaining_capital < 0.0f) [[unlikely]]
```

File: exchange/src/client_manager/client_manager.cpp [hunk 9]
```diff
@@ -3,58 +3,67 @@
...
         return messages::SIDE::BUY;
 
-    if (match.seller_uid != "SIMULATED" && insufficient_holdings) [[unlikely]]
+    if (match.seller_id != "SIMULATED" && insufficient_holdings) [[unlikely]]
```
Comment (egelja) [inline exchange/src/client_manager/client_manager.cpp:66]:
```suggestion
    // Simulated traders get infinite holdings
    if (match.seller_id != "SIMULATED" && remaining_holdings < 0.0f) [[unlikely]]
```

File: exchange/src/client_manager/client_manager.cpp [hunk 10]
```diff
@@ -63,52 +72,52 @@ ClientManager::validate_match(const messages::Match& match) const
...
+    for (auto& [id, user] : users) {
         if (!user.contains("latestAlgoId"))
             continue;
-        add_client(uid, user["latestAlgoId"].get<std::string>(), false);
+        add_client(id, user["latestAlgoId"].get<std::string>(), false);
     }
 }
 
 void
 ClientManager::add_client(
```
Comment (egelja) [inline exchange/src/client_manager/client_manager.cpp:83]: should be in header

File: exchange/src/client_manager/client_manager.cpp [hunk 11]
```diff
@@ -63,52 +72,52 @@ ClientManager::validate_match(const messages::Match& match) const
...
+    const std::string& id, const std::string& algo_id, bool is_local_algo
 )
 {
-    clients[uid] = Client{uid, algo_id, false, is_local_algo, STARTING_CAPITAL, {}};
+    clients[id] = Client{id, {}, algo_id, false, is_local_algo, STARTING_CAPITAL, {}};
 }
 
 void
-ClientManager::add_client(const std::string& uid, const std::string& algo_id)
+ClientManager::add_client(const std::string& id, const std::string& algo_id)
```
Comment (egelja) [inline exchange/src/client_manager/client_manager.cpp:91]: should be in header

File: exchange/src/client_manager/client_manager.cpp [hunk 12]
```diff
@@ -63,52 +72,52 @@ ClientManager::validate_match(const messages::Match& match) const
...
 {
-    clients[uid] = Client{uid, algo_id, false, false, STARTING_CAPITAL, {}};
+    clients[id] = Client{id, {}, algo_id, false, false, STARTING_CAPITAL, {}};
 }
 
 void
-ClientManager::modify_capital(const std::string& uid, float change_in_capital)
+ClientManager::modify_capital(const std::string& id, float change_in_capital)
```
Comment (egelja) [inline exchange/src/client_manager/client_manager.cpp:97]: should be in header

File: exchange/src/client_manager/client_manager.cpp [hunk 13]
```diff
@@ -63,52 +72,52 @@ ClientManager::validate_match(const messages::Match& match) const
...
+    if (!user_exists(id))
         return;
 
-    clients[uid].capital_remaining += change_in_capital;
+    clients[id].capital_remaining += change_in_capital;
 }
 
 float
-ClientManager::get_capital(const std::string& uid) const
+ClientManager::get_capital(const std::string& id) const
```
Comment (egelja) [inline exchange/src/client_manager/client_manager.cpp:106]: should be in header

File: exchange/src/client_manager/client_manager.cpp [hunk 14]
```diff
@@ -63,52 +72,52 @@ ClientManager::validate_match(const messages::Match& match) const
...
+    if (!user_exists(id))
         return 0.0f;
 
-    return clients.at(uid).capital_remaining;
+    return clients.at(id).capital_remaining;
 }
 
 void
-ClientManager::set_active(const std::string& uid)
+ClientManager::set_active(const std::string& id)
```
Comment (egelja) [inline exchange/src/client_manager/client_manager.cpp:115]: should be in header

File: exchange/src/matching/engine/engine.cpp [hunk 15]
```diff
@@ -139,11 +139,11 @@ Engine::attempt_matches(
...
 
-        std::string buyer_uid = buy_order.client_uid;
-        std::string seller_uid = sell_order.client_uid;
+        std::string buyer_id = buy_order.client_id;
+        std::string seller_id = sell_order.client_id;
 
-        Match toMatch = Match{sell_order.ticker, buyer_uid,      seller_uid,
-                              aggressive_side,   price_to_match, quantity_to_match};
+        Match toMatch{sell_order.ticker, aggressive_side, price_to_match,
+                      quantity_to_match, buyer_id,        seller_id};
```
Comment (egelja) [inline exchange/src/matching/engine/engine.cpp:146]:
```suggestion
                      quantity_to_match, buyer_id,        seller_id,
                      };
```
and reformat

**Review by egelja — CHANGES_REQUESTED** [review]

File: exchange/src/client_manager/client_manager.cpp (hunk 1 above)
Comment (stevenewald) [reply exchange/src/client_manager/client_manager.cpp:9]: It's a private function so it's only used in the implementation file. Why put it in the header?

File: exchange/src/matching/engine/engine.cpp (hunk 15 above)
Comment (stevenewald) [reply exchange/src/matching/engine/engine.cpp:146]: Clang format isn't liking this

File: exchange/src/client_manager/client_manager.cpp (hunk 2 above)
Comment (stevenewald) [reply exchange/src/client_manager/client_manager.cpp:13]: It's only used once in program execution so I'm gonna keep it out for the sake of readability

File: exchange/src/client_manager/client_manager.cpp (hunk 3 above)
Comment (stevenewald) [reply exchange/src/client_manager/client_manager.cpp:27]: No, this is a const function and clients[id] modifies the member variable clients

File: exchange/src/client_manager/client_manager.cpp (hunk 4 above)
Comment (stevenewald) [reply exchange/src/client_manager/client_manager.cpp:36]: [id] removes the "const" qualifier from the function

**Pull request merged**